src/Backend/Modules/Commerce/Domain/Cart/Cart.php

```diff
@@ -503,7 +503,8 @@ private function calculateCartRules(): void
             }
 
             if ($cartRule->getReductionPrice() !== null) {
-                $total = $this->applyPercentageDiscount($cartRule->getReductionPrice() / $this->total);
+                $this->moneyFormatter = new MoneyFormatter();
+                $total = $this->applyPercentageDiscount($this->moneyFormatter->asFloat($cartRule->getReductionPrice()) / $this->moneyFormatter->asFloat($this->subTotal));
```

Comment:
Hmm MoneyPHP (https://github.com/moneyphp/money) is used, to prevent calculation errors that happen when using floats instead. So ideally we try to remove floats as many as possible... https://friends-of-forkcms.github.io/fork-cms-module-commerce/money-pattern/

The old code that was changed here and calculates the total was probably not using Money objects yet... 

Multiplication/division calculations can be done on the object itself (https://www.moneyphp.org/en/stable/features/operation.html#multiplication-division) or https://www.moneyphp.org/en/stable/features/operation.html#ratio-of

```php
$cartRule->getReductionPrice()->ratioOf($this->subTotal)
```

But the `applyPercentageDiscount` will also need a signature change to change float to Money...